homeassistant/components/nsw_fuel_station/config_flow.py

```diff
@@ -0,0 +1,202 @@
+                if (
+                    search_term.lower() in station.name.lower()
+                    or search_term.lower() in station.address.lower()
+                )
+            ]
+            if not self.stations:
+                errors["base"] = "no_matching_stations"
+            else:
+                return await self.async_step_select_station()
+
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(INPUT_SEARCH_TERM): str,
+                },
+            ),
+        )
+
+    async def async_step_select_station(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Show the station selector form."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            station_id = int(user_input[INPUT_STATION_ID])
+            self.selected_station = next(
+                station for station in self.stations if station.code == station_id
+            )
+            await self.async_set_unique_id(str(station_id))
+            self._abort_if_unique_id_configured()
+            if "base" not in errors:
+                return await self.async_step_select_fuel()
+
+        return self.async_show_form(
+            step_id="select_station",
+            errors=errors,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(INPUT_STATION_ID): SelectSelector(
+                        SelectSelectorConfig(
+                            options=[
+                                SelectOptionDict(
+                                    value=str(station.code),
+                                    label=f"{station.name} - {station.address}",
+                                )
+                                for station in self.stations
+                            ],
+                            mode=SelectSelectorMode.DROPDOWN,
+                        )
+                    ),
+                },
+            ),
+        )
+
+    async def async_step_select_fuel(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Show the fuel type selection form."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.selected_station.name,
+                data={
+                    CONF_STATION_ID: self.selected_station.code,
+                    CONF_FUEL_TYPES: user_input[INPUT_FUEL_TYPES],
+                },
+            )
+
+        valid_fuel_types = {
+            fuel_type: self.data.fuel_types.get(fuel_type, fuel_type)
+            for station_code, fuel_type in self.data.prices
+            if station_code == self.selected_station.code
+        }
+
+        if len(valid_fuel_types) < 1:
+            return self.async_abort(reason="no_fuel_types")
```

Comment:
is this something that can happen? Should we just omit these from the ones you can select?